mapproxy/cache/file.py

```diff
@@ -42,179 +42,20 @@ def __init__(self, cache_dir, file_ext, directory_layout='tc',
         self.cache_dir = cache_dir
         self.file_ext = file_ext
         self.link_single_color_images = link_single_color_images
```

Comment:
There's one feature which would be nice to "backport" from `FileCache`: `link_single_color_images`.
AFAIK the *REST API* of S3 doesn't provide such "symbolic link" feature, compared to the common Web endpoint which supports the aptly named "Web Page Redirect": http://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html
But for compatibility reason I think we can reuse the same `x-amz-website-redirect-location` property / header.
The code would check if the property is set when fetching an object from S3 and retrieve the "real" object in that case.